test/test_blockarrayinterface.jl

```diff
@@ -21,22 +21,21 @@ struct PartiallyImplementedBlockVector <: AbstractBlockArray{Float64,1} end
         @test err isa ErrorException && err.msg == "setblock! for PartiallyImplementedBlockVector is not implemented"
     end
     try
-        BlockArrays.blocksize(A, 2)
+        BlockArrays.axes(A)
     catch err
-        @test err isa ErrorException && err.msg == "blocksizes for PartiallyImplementedBlockVector is not implemented"
+        @test err isa ErrorException && err.msg == "axes for PartiallyImplementedBlockVector is not implemented"
```

Comment:
Nothing to do this with this PR, but maybe use

```julia
@test_throws(
    ErrorException("axes for PartiallyImplementedBlockVector is not implemented"),
    BlockArrays.axes(A),
)
```

?

ATM, the test still passes even though the `try` part doesn't throw.